nodebuilder/node/uptime.go

```diff
@@ -0,0 +1,83 @@
+// This file defines UptimeMetrics relative to the nodebuilder package.
+package node
+
+import (
+	"context"
+	"time"
+
+	"go.opentelemetry.io/otel/metric/global"
+	"go.opentelemetry.io/otel/metric/instrument"
+	"go.opentelemetry.io/otel/metric/instrument/asyncfloat64"
+	"go.opentelemetry.io/otel/metric/instrument/syncfloat64"
+)
+
+// UptimeMetrics is a struct that records
+// 1. node start time: the timestamp when the node was started
+// 2. node up time: total time the node has been running
+//
+// the node start time is recorded imperatively when RecordNodeStartTime is called
+// whereas the node up time is recorded periodically upon callback recalling (re-mettering from optl)
+type UptimeMetrics struct {
+	// nodeStartTS is the timestamp when the node was started.
+	nodeStartTS syncfloat64.UpDownCounter
+
+	// totalNodeUptime is the total time the node has been running.
+	totalNodeUptime asyncfloat64.Gauge
```

Comment:
This could benefit from using a single counter. Add 1 every second node is running. It will give you run indicater + sum could be used as uptime in seconds later. Gauge is bad here, the value could be calculated by aggregate. 

Start time on the other hand should be a Gauge. Do you even need start time? You can do uptime calculation with it.

Reply:
Turning this into a counter adding 1 every other second is a great idea! Thank you
On the other hand, I think we need to record the node start time timestamp to achieve the uptime index formula from Mustafa.